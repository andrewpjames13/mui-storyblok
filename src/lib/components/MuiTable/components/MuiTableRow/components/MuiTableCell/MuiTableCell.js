/* eslint-disable max-len */
import React from 'react';
import PropTypes from 'prop-types';
import { TableCell } from '@material-ui/core';

import StoryBlok from '../../../../../../utils/Storyblok';
import MuiTypography from '../../../../../MuiTypography/MuiTypography';

const MuiTableCell = ({
  content,
  rootClass,
  align,
  padding,
  size,
  sortDirection,
}) => {
  const styles = StoryBlok.arrayToMuiStyles(rootClass);

  return (
    <TableCell
      className={styles.root}
      align={align}
      padding={padding}
      size={size}
      sortDirection={sortDirection}
    >
      <MuiTypography {...content[0]} />
    </TableCell>
  );
};

export default MuiTableCell;

MuiTableCell.propTypes = {
  /**
   * stroyblok multiselect of css classes
   * Mui Override or extend the styles applied to the component.
   */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /**
   * mui prop: 'center' | 'inherit' | 'justify' | 'left' | 'right'
   * Set the text-align on the table cell content
   */
  align: PropTypes.string,
  /**
   * mui prop: 'checkbox' | 'default' | 'none'
   * Sets the padding applied to the cell.
   */
  padding: PropTypes.string,
  /**
   * mui prop: 'small' | 'medium'
   * Specify the size of the cell
   */
  size: PropTypes.string,
  /**
   * mui prop: 'asc' | 'desc' | false
   * Set aria-sort direction.
   */
  sortDirection: PropTypes.string,
  /**
   * Content passed to render
   * components: MuiTypography,
   */
  content: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,
};

MuiTableCell.defaultProps = {
  align: 'inherit',
  padding: 'default',
  size: 'medium',
  sortDirection: false,
  rootClass: [],
};
