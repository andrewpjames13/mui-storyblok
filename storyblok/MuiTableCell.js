module.exports = {
  name: 'MuiTableCell',
  display_name: null,
  updated_at: '2020-04-20T17:46:56.399Z',
  schema: {
    content: {
      type: 'bloks',
      restrict_components: true,
      component_whitelist: [
        'MuiTypography',
      ],
      maximum: '1',
      required: true,
    },
    rootClass: {
      type: 'options',
    },
    align: {
      type: 'option',
      use_uuid: true,
      description: "   * mui prop: 'center' | 'inherit' | 'justify' | 'left' | 'right'\n   * Set the text-align on the table cell content",
      options: [
        {
          value: 'center',
          name: 'center',
        },
        {
          value: 'inherit',
          name: 'inherit',
        },
        {
          value: 'justify',
          name: 'justify',
        },
        {
          value: 'left',
          name: 'left',
        },
        {
          value: 'right',
          name: 'right',
        },
      ],
      default_value: 'center',
    },
    padding: {
      type: 'option',
      use_uuid: true,
      options: [
        {
          value: 'checkbox',
          name: 'checkbox',
        },
        {
          value: 'default',
          name: 'default',
        },
        {
          value: 'none',
          name: 'none',
        },
      ],
      description: "   * mui prop: 'checkbox' | 'default' | 'none'\n   * Sets the padding applied to the cell.",
      default_value: 'default',
      required: true,
    },
    size: {
      type: 'option',
      default_value: 'medium',
      description: "   * mui prop: 'small' | 'medium'\n   * Specify the size of the cell",
      use_uuid: true,
      options: [
        {
          value: 'small',
          name: 'small',
        },
        {
          value: 'medium',
          name: 'medium',
        },
      ],
      required: true,
    },
    sortDirection: {
      type: 'option',
      description: "   * mui prop: 'asc' | 'desc' | false\n   * Set aria-sort direction.",
      default_value: 'asc',
      required: true,
      use_uuid: true,
      options: [
        {
          value: 'asc',
          name: 'asc',
        },
        {
          value: 'desc',
          name: 'desc',
        },
        {
          value: 'false',
          name: 'false',
        },
      ],
    },
  },
  image: null,
  preview_field: null,
  is_root: false,
  preview_tmpl: null,
  is_nestable: true,
  all_presets: [],
  preset_id: null,
  real_name: 'MuiTableCell',
  component_group_uuid: null,
};
