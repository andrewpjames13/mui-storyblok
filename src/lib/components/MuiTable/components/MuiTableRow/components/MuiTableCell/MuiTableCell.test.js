import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import MuiTableCell from './MuiTableCell';

function setup() {
  const props = {
    align: 'inherit',
    padding: '1px',
    size: 'medium',
    sortDirection: 'asc',
    content: [{
      align: 'inherit',
      color: 'initial',
      width: '100%',
      height: '100%',
      content: [{
        text: 'Single table row',
        component: 'MuiText',
      }],
      variant: 'body1',
      component: 'MuiTypography',
    }],
  };
  const comp = shallow(<MuiTableCell {...props} />);
  return { comp, props };
}

describe('<MuiTableCell />', () => {
  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(
      <MemoryRouter>
        <MuiTableCell {...props} />
      </MemoryRouter>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render MuiTableCell', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });
});
