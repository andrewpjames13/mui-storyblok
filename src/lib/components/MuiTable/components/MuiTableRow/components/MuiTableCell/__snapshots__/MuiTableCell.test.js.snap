// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiTableCell /> snapshot 1`] = `
<td
  aria-sort="ascending"
  className="MuiTableCell-root makeStyles-root-2"
>
  <div
    className="MuiTypography-root makeStyles-root-3 MuiTypography-body1"
  >
    <span
      className="MuiTypography-root makeStyles-root-4 MuiTypography-body1"
    >
       Single table row 
    </span>
  </div>
</td>
`;
